backend/src/main/java/ca/bc/gov/app/converters/SubmissionMatchDetailEntityBeforeConvert.java

```diff
@@ -17,15 +17,28 @@
 import org.springframework.stereotype.Component;
 import reactor.core.publisher.Mono;
 
+/**
+ * This class is responsible for converting SubmissionMatchDetailEntity before and after it is saved in the database.
+ * It implements BeforeConvertCallback and AfterConvertCallback interfaces.
+ */
 @Component
 @RequiredArgsConstructor
 @Slf4j
 public class SubmissionMatchDetailEntityBeforeConvert
     implements BeforeConvertCallback<SubmissionMatchDetailEntity>,
     AfterConvertCallback<SubmissionMatchDetailEntity> {
 
+  // ObjectMapper provides functionality for reading and writing JSON
   private final ObjectMapper mapper;
 
+  /**
+   * This method is called before the entity is saved in the database.
+   * It converts the matchers field of the entity to a JSON string.
+   *
+   * @param entity The entity that is about to be saved in the database.
+   * @param table The table where the entity will be saved.
+   * @return The entity with the converted matchers field.
```

Comment:
Should we say:
@<!-- -->return **A Publisher that emits** the entity with the converted matchers field.
instead?
I got this by looking at another class in this same PR.

Reply:
That makes complete sense. Updated the docs